src/requests/process-method.ts

```diff
@@ -71,8 +71,17 @@ function getParamSeparation(paramGroups: Dictionary<Parameter[]>): string[] {
     let def: string;
 
     if (groupName === 'query') {
-      def = 'const queryParams = new HttpParams();\n';
-      def += _.map(group, p => `queryParams.append('${p.name}', JSON.stringify(params.${p.name}));`).join('\n');
+      def = 'const queryParams = new HttpParams()\n';
+      def += _.map(group,
+              p => {
+          if (p.type === 'string') {
+            return `.set('${p.name}', params.${p.name})`;
+          } else {
+            return `.set('${p.name}', JSON.stringify(params.${p.name}))`;
+          }
+        },
+      ).join('\n');
```

Comment:
Not sure, are `.set` invocations indented to proper level?

Reply:
I see it's not. There's also one thing more important than sparing `JSON.stringify` for string types (btw, I would test real type, not swagger type), not sending undefined params `?param1=undefined&param2=undefined`, so I suggest the following `if` block:

```typescript
if (groupName === 'query') {
  def = 'let queryParams = new HttpParams();\n';
  group.forEach(p => {
    def +=
    `if (params.${p.name} !== undefined) ` +
    `queryParams = queryParams.set('${p.name}', JSON.stringify(params.${p.name}));\n`;
  });

  return def;
}
```

Reply:
JSON.stringify must not be used on string, otherwise the resulting query param look like this `stringField=%22example-producer%22` instead of `stringField=example-producer`, i.e. the quote signs are added.